src/test/java/seedu/duke/commands/DeleteAccountCommandTest.java

```diff
@@ -0,0 +1,101 @@
+            DeleteAccountCommand command = new DeleteAccountCommand("delete-account JPY");
+            assertThrows(InvocationTargetException.class, () -> method.invoke(command, "JP"));
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void getCurrency_validCurrencyProvided_shouldReturnCorrespondingCurrency() {
+        try {
+            Method method = DeleteAccountCommand.class.getDeclaredMethod("getCurrency", String.class);
+            method.setAccessible(true);
+            DeleteAccountCommand command = new DeleteAccountCommand("delete-account JPY");
+            assertEquals(Currency.JPY, method.invoke(command, "JPY"));
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void processCommand_commandLessThanTwoWords_shouldThrowException() {
+        try {
+            Method method = AddCommand.class.getDeclaredMethod("processCommand");
+            method.setAccessible(true);
+            DeleteAccountCommand command = new DeleteAccountCommand("delete-account");
+            assertThrows(IllegalArgumentException.class, () -> method.invoke(command));
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_accountNotEmpty_shouldThrowException() {
+        try {
+            AccountList account = new AccountList();
+            account.addAccount(Currency.KRW, 1000);
+            DeleteAccountCommand command = new DeleteAccountCommand("delete-account KRW");
+            Ui ui = new Ui();
+            command.execute(ui, account);
+
+            // Account should not be removed if account not empty
+            account.getAccount(Currency.KRW);
+
+        } catch (Exception e) {
+            fail();
+        }
+    }
+    @Test
+    public void deleteAccount_nonExistCurrency_shouldThrowException() {
+        try {
+            Method method = AccountList.class.getDeclaredMethod(("deleteAccount"), Currency.class);
+            method.setAccessible(true);
+            AccountList account = new AccountList();
+            account.addAccount(Currency.KRW, 1000);
+            assertThrows(InvocationTargetException.class, () -> method.invoke(account, Currency.JPY));
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_correctInputProvided_shouldDeleteAccount() {
+        try {
+            AccountList account = new AccountList();
+            account.addAccount(Currency.KRW, 0);
+            DeleteAccountCommand command = new DeleteAccountCommand("delete-account KRW");
+            Ui ui = new Ui();
+            command.execute(ui, account);
+
+            // fail test if account not removed
+            try {
+                account.getAccount(Currency.KRW);
+                fail();
+            }catch(Exception e) {
+                System.out.print("");
+            }
+        } catch (Exception e) {
+            fail();
+        }
```

Comment:
I think the correct test should be `assertThrows(NoAccountExistException.class, ()->account.getAccount(Currency.KRW));`